File: elixir/lasagna/lib/lasagna.ex
defmodule Lasagna do
  def expected_minutes_in_oven() do
    40
  end

  def remaining_minutes_in_oven(minutes) do
    expected_minutes_in_oven() - minutes
  end

  def preparation_time_in_minutes(layer_count) do
    2 * layer_count
  end

  def total_time_in_minutes(layer_count, elapsed_time) do
    preparation_time_in_minutes(layer_count) + elapsed_time
  end

  def alarm() do 
    "Ding!"
  end
end

File: java/lasagna/src/main/java/Lasagna.java
public class Lasagna {
    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int elapsedCookingTimeInMinutes) {
        return expectedMinutesInOven() - elapsedCookingTimeInMinutes;
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * 2;
    }

    public int totalTimeInMinutes(int numberOfLayers, int elapsedCookingTimeInMinutes) {
        return preparationTimeInMinutes(numberOfLayers) + elapsedCookingTimeInMinutes;
    }
}
